File: android-app/app/src/main/java/tech/callsafe/business/utils/DeviceUtils.kt
package tech.callsafe.business.utils

import android.content.Context

// Global utility function for device ID
fun getUniqueDeviceId(context: Context): String {
    val sharedPreferences = context.getSharedPreferences("CallSafePrefs", Context.MODE_PRIVATE)
    val existingDeviceId = sharedPreferences.getString("device_id", null)
    
    return if (existingDeviceId != null) {
        existingDeviceId
    } else {
        val newDeviceId = java.util.UUID.randomUUID().toString()
        sharedPreferences.edit().putString("device_id", newDeviceId).apply()
        newDeviceId
    }
}

File: flutter/ios/Runner/Utils/DeviceUtils.swift
import Foundation
import UIKit

class DeviceUtils {
    static func getUniqueDeviceId() -> String {
        let defaults = UserDefaults.standard
        let key = "device_id"

        if let existingDeviceId = defaults.string(forKey: key) {
            return existingDeviceId
        }

        // Generate new device ID
        let newDeviceId = UUID().uuidString
        defaults.set(newDeviceId, forKey: key)

        return newDeviceId
    }
}
